### PR TITLE
chore: setup TypeScript type checking

### PR DESCRIPTION
**Pull request by bajtos**
1. Configure TypeScript checking (locally and as part of the CI)
2. Fix TypeScript errors

See https://github.com/filecoin-station/roadmap/issues/80

The content of `tsconfig.json` is from station-core:
https://github.com/filecoin-station/core/blob/01e9a9f6169c6df5dc152bf7d89b04bade3e5d70/tsconfig.json)

Out of scope/follow-up issues:
- https://github.com/filecoin-station/on-contract-event/issues/2

### CODE REVIEW

File: bin/spark-evaluate.js [hunk 1]
```diff
@@ -39,7 +39,10 @@ const signer = ethers.Wallet.fromPhrase(WALLET_SEED, provider)
 console.log(
   'Wallet address:',
   signer.address,
-  newDelegatedEthAddress(signer.address, 'f').toString()
+  newDelegatedEthAddress(
+    /** @type {any} */(signer.address),
+    /** @type {import('@glif/filecoin-address').CoinType} */ ('f')
```
Comment (juliangruber) [inline bin/spark-evaluate.js:44]: can we directly pass this instead?

https://github.com/glifio/modules/blob/796cbf39c7e419f62e5c4ddf43f61db2bd14c2bc/packages/filecoin-address/src/coinType.ts#L8

https://github.com/glifio/modules/blob/796cbf39c7e419f62e5c4ddf43f61db2bd14c2bc/packages/filecoin-address/src/index.ts#L9

https://github.com/glifio/modules/blob/796cbf39c7e419f62e5c4ddf43f61db2bd14c2bc/packages/filecoin-address/src/index.ts#L27

```js
coinType.MAIN
```

**Review by juliangruber — COMMENTED** [review]
As far as I understand typescript, lgtm, and thank you for kicking this off!

File: bin/spark-evaluate.js (hunk 1 above)
Comment (bajtos) [reply bin/spark-evaluate.js:44]: Great suggestion! Applied.

**Comment by bajtos** [comment]
@juliangruber added a few more commits to fix the remaining issue and address your comment.

The PR is ready for final review and landing.

File: index.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import assert from 'node:assert'
 import * as Sentry from '@sentry/node'
 import { preprocess } from './lib/preprocess.js'
 import { evaluate } from './lib/evaluate.js'
+// TODO: implement typings - see https://github.com/filecoin-station/on-contract-event/issues/2
```
Comment (juliangruber) [inline index.js:5]: I think we should switch to ethers@6 instead

**Review by juliangruber — APPROVED** [review]

**Pull request merged**